File: LLM-Translations/GPT-4/codenet/C/Go/s065149518.go

package main

import (
	"fmt"
)

const MAX = 21

var n, x, y int
var fld [MAX][MAX]int

func main() {
	var m int
	for {
		_, err := fmt.Scan(&n)
		if err != nil || n == 0 {
			break
		}
		init(n)
		fmt.Scan(&m)
		move(m)
		if n == 0 {
			fmt.Println("Yes")
		} else {
			fmt.Println("No")
		}
	}
}

func init(i int) {
	var a, b int
	x, y = 10, 10
	fld = [MAX][MAX]int{}
	for i > 0 {
		fmt.Scan(&a, &b)
		fld[b][a] = 1
		i--
	}
}

func move(i int) {
	var a, d int
	var buf string
	for i > 0 {
		fmt.Scan(&buf, &a)
		switch buf {
		case "N":
			d = 0
		case "E":
			d = 1
		case "S":
			d = 2
		case "W":
			d = 3
		}
		for a > 0 {
			if d%2 == 0 {
				if d == 0 {
					y++
				} else {
					y--
				}
			} else {
				if d == 1 {
					x++
				} else {
					x--
				}
			}
			if fld[y][x] == 1 {
				fld[y][x] = 0
				n--
			}
			a--
		}
		i--
	}
}

// 

File: LLM-Translations/GPT-4/codenet/C/Python/s065149518.py

def init(i):
    global x, y, fld
    x = y = 10
    fld = [[0 for _ in range(MAX)] for _ in range(MAX)]
    for _ in range(i):
        a, b = map(int, input().split())
        fld[b][a] = 1

def move(i):
    global x, y, fld, n
    for _ in range(i):
        buf, a = input().split()
        a = int(a)
        if buf == 'N':
            d = 0
        elif buf == 'E':
            d = 1
        elif buf == 'S':
            d = 2
        elif buf == 'W':
            d = 3

        for _ in range(a):
            if d % 2:
                if d == 1:
                    x -= 1
                else:
                    x += 1
            else:
                if d == 2:
                    y -= 1
                else:
                    y += 1

            if fld[y][x]:
                fld[y][x] = 0
                n -= 1

def main():
    global n
    while True:
        n = int(input())
        if n == 0:
            break
        init(n)
        m = int(input())
        move(m)
        print("No" if n else "Yes")

if __name__ == "__main__":
    MAX = 21
    main()

#
